### PR TITLE
PWGHF: Improved MC-reco mode of DDbar correlation code

### PR DESCRIPTION
**Pull request by fcolamar**
Allows mimicking the data-like mode when running the D0D0bar and D+D- codes on MC-reco mode (which is he only possible mode runnable on hyperloop at the moment, since there's no support of global flags).
This was already possible at the level of the correlation distributions (splitting signal and background contributions in different objects and saving both), now the same is done also for the invariant mass plots needed for per-trigger normalisation.

### CODE REVIEW

**Comment by fcolamar** [comment]
As asked by @ginnocen I now set the status from draft PR to PR

File: Analysis/Tasks/PWGHF/HFCorrelatorD0D0bar.cxx [hunk 1]
```diff
@@ -194,8 +194,12 @@ struct HfCorrelatorD0D0barMcRec {
...
+    registry.add("hMassD0MCRecRefl", "D0 reflection candidates - MC reco;inv. mass (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {(std::vector<double>)bins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hMassD0MCRecBkg", "D0 background candidates - MC reco;inv. mass (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {(std::vector<double>)bins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hMassD0barMCRecSig", "D0bar signal candidates - MC reco;inv. mass D0 only (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {(std::vector<double>)bins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hMassD0barMCRecRefl", "D0bar reflection candidates - MC reco;inv. mass D0 only (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {(std::vector<double>)bins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hMassD0barMCRecBkg", "D0bar background candidates - MC reco;inv. mass D0 only (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {(std::vector<double>)bins, "#it{p}_{T} (GeV/#it{c})"}}});
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/HFCorrelatorD0D0bar.cxx]: Why does the axis label say "inv. mass D0 only", whereas the histograms are filled with `InvMassD0bar`?

File: Analysis/Tasks/PWGHF/HFCorrelatorDplusDminus.cxx [hunk 2]
```diff
@@ -192,8 +192,10 @@ struct HfCorrelatorDplusDminusMcRec {
...
-    registry.add("hMassDminusMCRec", "Dplus,Dminus candidates - MC reco;inv. mass D0 only (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {(std::vector<double>)bins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hMassDplusMCRecSig", "Dplus signal candidates - MC reco;inv. mass (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {(std::vector<double>)bins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hMassDminusMCRecSig", "Dminus signal candidates - MC reco;inv. mass D0 only (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {(std::vector<double>)bins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hMassDplusMCRecBkg", "Dplus background candidates - MC reco;inv. mass (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {(std::vector<double>)bins, "#it{p}_{T} (GeV/#it{c})"}}});
+    registry.add("hMassDminusMCRecBkg", "Dminus background candidates - MC reco;inv. mass D0 only (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH2F, {{120, 1.5848, 2.1848}, {(std::vector<double>)bins, "#it{p}_{T} (GeV/#it{c})"}}});
```
Comment (vkucera) [inline Analysis/Tasks/PWGHF/HFCorrelatorDplusDminus.cxx]: "D0 only" does not seem to fit here.

Comment (fcolamar) [reply Analysis/Tasks/PWGHF/HFCorrelatorDplusDminus.cxx]: Typo, fixed

File: Analysis/Tasks/PWGHF/HFCorrelatorD0D0bar.cxx (hunk 1 above)
Comment (fcolamar) [reply Analysis/Tasks/PWGHF/HFCorrelatorD0D0bar.cxx]: Typo, fixed

**Comment by fcolamar** [comment]
@ginnocen @vkucera do you know where this pragma check failure comes from? I really just changed few histogram labels since last commit

**Comment by vkucera** [comment]
> @ginnocen @vkucera do you know where this pragma check failure comes from? I really just changed few histogram labels since last commit

It's coming from some obsolete files in your branch that are fixed in `dev`. You need to rebase your branch.

**Review by vkucera — APPROVED** [review]

**Comment by fcolamar** [comment]
Hi @vkucera, @ginnocen, can this still be merged or do I need to push the commit with the merge with the rebased dev and redo the tests?

**Comment by vkucera** [comment]
@jgrosseo @iarsene Can you please merge this one? The pragma test failure is coming from unrelated files which are not part of the PR.

**Pull request merged**